readthedocs/analytics/tasks.py

```diff
@@ -81,4 +80,7 @@ def delete_old_page_counts():
     """
     retention_days = settings.RTD_ANALYTICS_DEFAULT_RETENTION_DAYS
     days_ago = timezone.now().date() - timezone.timedelta(days=retention_days)
-    return PageView.objects.filter(date__lt=days_ago).delete()
+    return PageView.objects.filter(
+        date__lt=days_ago,
+        date__gt=days_ago - timezone.timedelta(days=90),
```

Comment:
Seems this should be dynamic, and based on `retention_days`?

Reply:
It's based on `days_ago`, which is based on `retention_days`. I think it's fine.

It will delete `PageViews` older than `retention_days` and newer than `retention_days - 90 days`.